wikispeedruns/achievement_functions.py

```diff
@@ -145,10 +145,13 @@ def around_the_world_in_80_seconds(single_run_data: Dict[str, Any], single_run_a
             break
 
         article = path[i]["article"]
-        current_map[article] -= 1
-        if current_map[article] == 0:
-            del current_map[article]
-    
+        i += 1
+
+        if article in article_set:
```

Comment:
I didn't take a close look at this before, but this definitely seems a bit complicated. Could you add something in explaining it? From my impression, you're trying to find the subarray within the path that contains all 7 articles, but with minimum time? `i` is like a left index and `j` a right index?